tests/engines/registries/unit/services/share-search-test.ts

```diff
@@ -80,4 +82,18 @@ module('Registries | Unit | Service | share-search', hooks => {
         assert.equal(registrations[0].contributors[0].name, 'Graham > Berlin');
         assert.equal(registrations[0].contributors[1].name, 'Nicole < Grant');
     });
+
+    test('recognizes all OSF source envs', function(this: TestContext, assert) {
+        const service = this.owner.lookup('service:share-search') as ShareSearch;
+        [
+            'https://osf.io/w4yhb/',
+            'https://test.osf.io/w4yhb/',
+            'https://staging.osf.io/w4yhb/',
+            'https://staging2.osf.io/w4yhb/',
+            'https://staging3.osf.io/w4yhb/',
```

Comment:
Could you add a few cases with multiple identifiers, to check it *doesn't* match SHARE urls? e.g.
```ts
[{
  input: ['https://staging3.osf.io/w4yhb/', 'https://staging-share.osf.io/w4yhb/'],
  output: 'https://staging3.osf.io/w4yhb/',
}, {
  input: ['https://staging-share.osf.io/w4yhb/', 'https://staging2.osf.io/w4yhb/'],,
  output: 'https://staging2.osf.io/w4yhb/',
}, {
...